ios/FluentUI/Table View/TableViewCellFileAccessoryView.swift

```diff
@@ -0,0 +1,383 @@
+    fileprivate let useAppPrimaryColor: Bool
+}
+
+// MARK: - TableViewCellFileAccessoryView
+
+/// Class that represents a table view cell accessory view representing a file or folder.
+@objc (TableViewCellFileAccessoryView)
+open class TableViewCellFileAccessoryView: UIView {
+    /// The date will be displayed in a friendly format in the accessory view's first column.
+    @objc public var date: Date? {
+        didSet {
+            updateDateLabel()
+
+            if tableViewCell != nil && (date == nil && oldValue != nil) || (oldValue == nil && date != nil) {
+                updateLayout()
+            }
+        }
+    }
+
+    /// Set to true to display the shared status in the accessory view's second column, false otherwise.
+    @objc public var showSharedStatus: Bool = false {
+        didSet {
+            if tableViewCell != nil && oldValue != showSharedStatus {
+                updateLayout()
+            }
+        }
+    }
+
+    /// Set to true to indicate that the document is shared with others, false otherwise.
+    @objc public var isShared: Bool = false {
+        didSet {
+            updateSharedStatus()
+        }
+    }
+
+    /// Actions to display in the accessory view's third column.
+    /// Depending on the cell's width, some actions may be hidden.
+    /// A maximum of 4 actions will be displayed.
+    /// Actions with the smallest index in the array will have a higher priority for being displayed.
+    @objc public var actions: [FileAccessoryViewAction] = [] {
+        didSet {
+            if tableViewCell != nil && !oldValue.elementsEqual(actions) {
+                updateLayout()
+            }
+        }
+    }
+
+    @objc public weak var tableViewCell: TableViewCell? {
+        didSet {
+            updateLayout()
+        }
+    }
+
+    @objc public override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        addSubview(columnStackView)
+
+        NSLayoutConstraint.activate([
+            columnStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            columnStackView.topAnchor.constraint(equalTo: topAnchor),
+            columnStackView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+
+        columnStackView.addArrangedSubview(actionsStackView)
+
+        updateSharedStatus()
+    }
+
+    @available(*, unavailable)
+    @objc public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if let tableViewCell = tableViewCell {
+            if !tableViewCell.bounds.equalTo(previousBounds) {
+                previousBounds = tableViewCell.bounds
+
+                updateLayout()
+            }
+        }
+    }
+
+    private struct Constants {
+        static let layoutBreakPoints: [CGFloat] = [424.0, 504.0, 584.0, 616.0, 752.0, 900.0, 1092.0]
+        static let maxVisibleActionCount: UInt8 = 4
```

Comment:
It could but we would always end up converting these to CGFloats. I don't think it's much of a memory concern to use CGFloat (64-bit) instead of something smaller like UInt16 for these numbers.